### ISSUE TITLE
exon feature coordinates start greater than end in the GTF

### ISSUE THREAD

**Issue by creaturemoon** [alevar/sam2gtf#1]
Hello,
  I performed SAM conversion to GTF using this tool. Reads were from ONT and aligned with reference genome using minimap2 to generate SAM. There is a read `08276814-5370-4ac2-abef-ecea721598b6` in the GTF where the starting position of one exon (second) is greater than the end. 
GTF about this read,

> 2L      SAM2GTF transcript      21060956        21156183        .       -       .       transcript_id "5053117"; read_name "08276814-5370-4ac2-abef-ecea721598b6";
2L      SAM2GTF exon    21060956        21060962        .       -       .       transcript_id "5053117"; read_name "08276814-5370-4ac2-abef-ecea721598b6";
2L      SAM2GTF exon    21066507        21066506        .       -       .       transcript_id "5053117"; read_name "08276814-5370-4ac2-abef-ecea721598b6";
2L      SAM2GTF exon    21154286        21154579        .       -       .       transcript_id "5053117"; read_name "08276814-5370-4ac2-abef-ecea721598b6";
2L      SAM2GTF exon    21154646        21154985        .       -       .       transcript_id "5053117"; read_name "08276814-5370-4ac2-abef-ecea721598b6";
2L      SAM2GTF exon    21155107        21155365        .       -       .       transcript_id "5053117"; read_name "08276814-5370-4ac2-abef-ecea721598b6";
2L      SAM2GTF exon    21155572        21155681        .       -       .       transcript_id "5053117"; read_name "08276814-5370-4ac2-abef-ecea721598b6";
2L      SAM2GTF exon    21155744        21156183        .       -       .       transcript_id "5053117"; read_name "08276814-5370-4ac2-abef-ecea721598b6";

SAM about this read,

> 08276814-5370-4ac2-abef-ecea721598b6    16      2L      21060956        60      87S7M269I5544N938I87779N128M2I98M1I8M1D59M66N15M1I18M3D25M1D28M1I29M1D29M1D1M1D9M1D2M3I57M2I115M1D1M1D1M121N153M1D96M1D8M206N59M2I37M1I4M2D8M62N178M1I43M3D76M1D98M2D39M        *       0       0       GGGGATTTTTTTTTTTTTTTTTTTTGTTTTTTTGTTTTAGTTTTTATTATTAACTTTATTTGTAGTTTTTTTTGTGGGAGATTTCCTTTGCCGCAGTAATTATCATTTTTTGCTTATATGGTAGCTATATAATTATATAGTGTATATCTAGTTGGATGACAAACATACATATACAGACAGACACACACACTGAAGTTGCACATTTAGTTAGCTCGAAATTAGATTTTGCTCTGTTGTTCTTATTGGTTGCTTTTTTTTGCATTTTGTTTATAGTGTATTCCTTACTTCGCTTCGCTCATCCTTTCTCCCCCCTCCCCCTGCTCCTACGGCTCCACCTCTCCTACTTTCTCCCATACTAGACACACAGACATTTAGTGGTATTTAAATATATATATATATCATATATCTAGCATTATCGCTATTGTTTACATATGTGAAAATTTGTTGGCTGATCGCGCATTATTTTAAACACACAATTACAATTACATTTATAACATACACAGATACACAGTAAATTTTATTTAACCATCTTCCTGCTGACATCCTACTAGTATCTCCCCATGCTTTCTCACATTGTCTTGGTTTTCTTTTTCTCTTATTTTTCTTTTTTTGTTTTGTTTTAGTTTTTAGTCATTTTCTTGTTCTTGCTGTGTTTGTCGTTGGCGCCCCCGCCTCGCCCCTCGAGATGGGTGGAGCGTTTGTGGGTAGGCGGATGGCAGGGAGGAGTAGACTGGCGGTGGCCGAGGCGGGGAGTGGGGGGCGGCGAGGGAGGTCTCACTCAATGGATTCAACGACAAGGGGGACGCACCACCACACAGAGAAAATACTTGACGCGTTTACATATATCACGGAAACACAAGCCATCGATACAAAAGAATATATTGTATATATAAAAATAAATATACCCTACTTAGCCAAGTTCTGTTGTCTCGCTCGCACACACAATCAGGAATCACTAATCACACACTGCACAATAGACCCAAGTATATTATATACTATATTATAAATATGTATATGCCTGGCTGCTGATCCTCATTTGAATGAATCGATCGATCGATGGGCTGCCTCCTGCTAACGCCTCCTGCCTCCTCCAGCTTCCTCCTGCTATCTTCTCTCTATGGCTATTACTAATATGCTCCTTAGTTCACCCCCAAAAGACACAAAGCACACATGGCACACATGGCACTGGCACATGGCCCACATTCATACACAAACGCACATTGATCTAAAGGGGGATAGGGATGTAGGGGGGGGATGATATATATAGATAAGATAAGGGGGTTGGATGGGGGATTTTTTTTTTTTTTTGCATTTACGAATTTATTGGTTTTGCTGTTATTTGCGAAATTACAAGGTGCTACACTAATTAAAAATTTCTAGACGATATAGAGAAATCCAAAGTCAAAATTGTACTAATCATAAATATTTGCACACAATGCACGCGGCCAGTTGCAGTGTAGCTATAGTCCTAAGGTGTACATCCCCTTCCACAGTTAGGAGCACTGAACATCAAACAGCTCGCAAATACCTTTCGTCGCTTTTAACGCAAAGTTGAAGTCTTCCTGCGGAGGATTCAGCTGCAGCAGCGGCGGGTTTTCTAGATTCGGATCGTACGGGATAAAATCGCCGCAAATGACCGGACAGATAATTTTGGAAATTATCTTGGCCGTCAGCTCGTTCTCCACGAAGTTTCATCTCATCTTTCATCCCGGGATGCACGAGTAACCGTGCTCGTCGGTGATCCATCCCCGCGATCCCGCTGGAGCGTTACCATCCTCGCTGGAGTGCGACGACGTCGAGGGTGAGGGCGTGTCCAGAGACGGTGATAGTCAAAGAACCCATCCACATCGTGCGGATTTGTAGTACTCTTTCCTTGGTTCGTCACCAGGCGAATGTCAATTGCGCTGCCATTGTCCAGCGACACGGCCAGCTGAGTGATTACGCTTGATATCTACTTCCTCGTCATAATTAGGTATTGTGAATACGGAATCATCTCCAAAGATGTCCAGCAGATCATCACGTGTCACATATGCATACGACCTATTCGAGGGATCCTGCATAACGTAGCGCAGATTGCGCTGTGCATATCTAGTTGCCTGTCTAGGTCATCCTCCAACATTTTCAAATGATTAGTCCGGCTACGTGCCAGGTCGTAGTTCTCTTGGTCCTTGGCATTGTTAAAGCCCCTCCGCGCCAGCGCACTAGGGAGCAGTGTCTGCCCTTATCAATTAGTCCAATGCCCTCTAAAACATATTGGTAATATCGTATATTCGGCGCTTCTGTACGTCCAAAGATTGGTTGCCGCTTTCAGGTCGATGGATCCTTCGTTGGCCTTCACGAGATCCACAGACTTCTGTGTGAGCAGGACCAGTGAGCCCACGGAGCGCTGTTGTCCAGGCTGCGATTGCGATTGTGAATTCGACGGAGACTTGATTGGCGTTGGAGGATCCGGTGATGTGTCCATGCTAACGGGTGTAAACTTCGTAGCTCTGCGACCGGACCGAAGTCTCAGCCGAATCTACCTTCATCATAGCCGAGGAGGTGGTGCCCGCTGCGGAGCTGTCTCTTTTAACAATACTCGCGGTTTTGCGCTTGTACATGTCCGCAATCGACTCGATGACTATACTCCTGTCCTTGTAACTCTTTGCGTGGCTGCACTCAATATGCAGCGATCAAAGAGCCGGAATTAATTGCACTTGAGCGAAAATTAAACGAAAATACGAATGTTGGCGCTAGG       %%&&'(('/+.1245689::8630,-+-.0587,0258;:968:6:8866<=><::98977791()))4-299=;::60.'''-'./014><87779?<<<<<<<==:92;98898:81<<<=;7-....70--)+5779<>?@EB??=?76666<;<<38+++,5;5:;;:;;;<?<=8888086:99:9;211///./:<>@<<@76++6653333+3:;;<00/008<<<AB?AA=6666>????>,,,10.9;@;85757304:======;::99;99;<@GF:43333....7<B<@?86>.99<:>A?>>=99466679:8333274333466678;?A@?7988+**++<:0000/:;>=><===:<98788<;998<88::;6;<<A?@ACC@?<<<<>:==;<<:::733323764*102221/0667740&85=>??<;:::<<=>=:::899::=;=:/>>>=>;::;66680///6562231133499;;;<8<20000::;9946:<@B@?A75563<<<<:=;<899997==<=><57?C>?;:421112875)(''')('%###$'2233341-05255574<9:779<111<<==;633458874:<;4331<9888:=<=<CCC>>;:::7663,)()7:9/...-234==65444-&&&&-02475334;::67604144555556500**''''&(((311.&.0/144442482;41,,--./-,+&&---.25588:<?@<75+*('(02344555333346677799885221/0///.-.-.../225567:9632,+,01.322211358642/*.///04454322334+21-+%%,,,./12225.-()+,*++7:;:<<<??<><;:<@A>@@<>7000.../4999;>=;66668<;;;<A>776667788687::-)(*/1211259<;:997754000''''*124488=433777779::@@116323+()1540////.19==<>=<<::;98888;;;9999:;<<D=5587449@?>@@DCFI<DEEC@@?==3::<;:::644445=>@@CC<@?A=;;;;=9<=A;;;;<EE=D@??:945*****+9<:987778@@A=><;::<<;===<;/.--.,,,&$$$:65-./0=A>@@?@@:B>65566;CC:72223====;;<=<=<::>>699<32-0/111=?<:,,,73*')++*(####$$$$##"#"$%'(,)%%&&'(((),+****))('&&%%%(&%##$$$$%$$%&((')*+./1345652/**++232/...278:<:7,,648;>;76678.++87531477600/.'''(27:88743699,,,,,,.33322877663446383<>8A545;997:<@?>><<9922:=>>><>=<::::;;77;;;88650//00555549<96346666877789:;<9898;:<<<>=:;98;;;:98332678>>>72////000000024431.,----*35445-)-32222236275311//..4445666/////4000-.2)))9=;755543>=<=;7643////05.--,+/3333/,*&%$$$%,-11334::9642/$$$%'%%&00123335766656**($$%)1233523112376676-+*)*1-,&%%%%'255678886673354486656661-,,,.266789;;:9;:99887;:87210.,,,)*'%$##%%&'%%$%&&&.../9999;;----.:;;;::841+*++-/678;;:51121/(&'845556:4-,,01888822225996<:0195978;:59987844333559;;;;934454322354556877766:<:6555567665789:@><22122788::9976431///-...+,,****,./22243222356676765677::6332222221222473..230&&30.---/154333355165322124321../24321/,,,,2124458876744444665663420//017876555640///0332111122123355676/*&&%%%&-//0/-.--,++.134/////002239:;897776654011004444322110//0135611122;6767221000156-----77798831111.....34555566:8978<9:76655567:65464121+))))))-..*,5524334436768-(()126;;9:768767766789:99899;<;::9565*))(&&&'..678::;94::45777;<:;=<9:==<?ACB>;:;77<:::55677?=<0/../672-,,,,-0065535665;::99899>?B==22226<@AA>=<://:76321001266.(''&'((4664.../167554336679650/)(()269::9:;:65430....//1/-//2200((((()(+488:;;78:97//,+*467988888333445-+*&((()--*),9962112344445566332211347<<894233366656:<:63333678887755577;<752/./0....,((0000102333333432878878765444457899985544477:8888876522213332565*)522...66+****32'&&&'22337779:72'&&***+(133338754/'875443444435666321       NM:i:1258       ms:i:1302       AS:i:1038       nn:i:0  ts:A:+  tp:A:P  cm:i:361
        s1:i:1270       s2:i:1086       de:f:0.0289     rl:i:100

Do you know how it was generated and how to handle it? Thanks for your help.

Kind Regards,

**Comment by alevar** [comment]
Hi! Coincidentally, I just discovered the same issue myself. The script was initially written for quick examination of PacBio reads which have significantly lower error rate. When I ran on ONT reds a few days ago - observed the same behavior. The issue comes from alignments containing spurious insertions within intronic segments of the cigar string: eg. 5N1I5N. This behavior was initially not accounted for (as are many other exceptions). I have pushed a fix for this particular issue just now. Hope it works for you!

Thank you,

AS
